src/components/atoms/ThresholdChart.tsx

```diff
@@ -29,59 +29,64 @@ const ThresholdChart: React.FC<ThresholdChartProps> = ({
     yLabel,
 }) => {
     // accessors
-    const date = (d: TimeEntry) => new Date(d.time).valueOf();
-    const value = (d: TimeEntry) => d.value;
+    const xValue = (d: ChartEntry): number => {
+        return typeof d.x === 'number' ? d.x : new Date(d.x).valueOf();
+    };
+
+    const yValue = (d: ChartEntry) => d.y;
 
     //bounds
     const xMax = width - margin.left - margin.right;
     const yMax = height - margin.top - margin.bottom;
 
     //scales
-    const timeScale = scaleTime({
-        domain: [Math.min(...data.map(date)), Math.max(...data.map(date))],
+    const xScaleValues = {
+        domain: [Math.min(...data.map(xValue)), Math.max(...data.map(xValue))],
         range: [0, xMax],
-    });
+    };
+
+    const xScale = typeof data[0].x === 'number' ? scaleLinear(xScaleValues) : scaleTime(xScaleValues);
 
-    let maxValue = Math.max(...data.map(value));
-    let minValue = Math.min(...data.map(value));
+    let maxValue = Math.max(...data.map(yValue));
+    let minValue = Math.min(...data.map(yValue));
 
     //prevents the threshold to be outside the graph.
     if (thresholdValue > maxValue) maxValue = thresholdValue + 1;
     //+1 is needed for the colour to work below the graph when the threshold is bigger than the highest value
     else if (thresholdValue < minValue) minValue = thresholdValue;
 
-    const valueScale = scaleLinear({
+    const yScale = scaleLinear({
         domain: [minValue, maxValue],
         range: [yMax, 0],
         nice: true, //Gives the graph more space, rounds the y axis to closest integer.
     });
 
-    const firstDate = date(data[0]);
+    const firstX = xValue(data[0]);
```

Comment:
Do we need to make this a variable?

Reply:
Not really. I guess it makes the code more readable in some way?